webapp/src/js/enketo/medic-xpath-extensions.js

```diff
@@ -162,6 +163,28 @@ const dateDiff = function (startDateObj, endDateObj, key) {
   return XPR.number(endDate.diff(startDate, key));
 };
 
+const luhn = function (number) {
+  number = getValue(number);
+  number = number.toString().trim().replace(/\s/g, '');
+
+  if (number.length !== 13 || !/^\d+$/.test(number)) {
+    return XPR.bool(false);
+  }
+
+  const digits = number.split('').map(Number);
+  const sum = digits.reduce((acc, digit, index) => {
```

Comment:
Our style guide advises to [avoid reduce](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/#avoid-reduce) for readability reasons. The wikipedia reference you linked has a typescript version which is more readable (IMO): https://en.wikipedia.org/wiki/Luhn_algorithm#TypeScript